CI: Nightly Migration to Github Actions

MericFeyz opened the pull request:
Workflow for releasing nightly builds, that triggers every night. It first tests the default branch by reusing the e2e test workflow, builds actual and pushes images to Dockerhub, and then updates the helm chart. 

E2E test workflow is now a reusable workflow. Jq install command is removed as it is present in setup script, and setting ```submodules``` parameter of ```actions/checkout``` to ```recursive``` lets us remove (comment it out in this case) the ```git submodule update``` line. 

Closes #553 

Signed-off-by: Meric <meric.feyzullahoglu@gmail.com>

wagmarcel reviewed (changes requested):
https://cbea.ms/git-commit/

wagmarcel reviewed (changes requested):
Second commit: Merge branch 'Open-IoT-Service-Platform:develop' into Iss_Nightly_Mig…
Looks like an internal merge of branches in your fork. Can you please make sure it is either squashed with the commit before or contains a proper commit message?

MericFeyz committed:
Implement nightly builds to the Github Actions

New workflow for releasing nightly builds, translated from CircleCI workflow.
E2E test workflow is now a reusable workflow, so that it can be called from the new workflow.
Jq install command is removed as it is present in setup script.
Submodules parameter of actions/checkout is utilized.

Closes: #553

Signed-off-by: Meric <meric.feyzullahoglu@gmail.com>
